sdmetrics/single_table/equalized_odds.py

```diff
@@ -0,0 +1,465 @@
+            ):
+                data[column] = data[column].astype('category')
+            elif column in data.columns and column_meta.get('sdtype') == 'datetime':
+                data[column] = pd.to_numeric(data[column], errors='coerce')
+
+        return data
+
+    @classmethod
+    def _train_classifier(cls, train_data, prediction_column_name):
+        """Train the XGBoost classifier."""
+        train_data = train_data.copy()
+        train_target = train_data.pop(prediction_column_name)
+
+        try:
+            from xgboost import XGBClassifier
+        except ImportError:
+            raise ImportError(
+                'XGBoost is required but not installed. Install with: pip install sdmetrics[xgboost]'
+            )
+
+        classifier = XGBClassifier(enable_categorical=True)
+        classifier.fit(train_data, train_target)
+
+        return classifier
+
+    @classmethod
+    def _compute_prediction_counts(cls, predictions, actuals, sensitive_values):
+        """Compute prediction counts for each sensitive group."""
+        results = {}
+
+        for sensitive_val in [True, False]:
+            mask = sensitive_values == sensitive_val
+            if not mask.any():
+                # No data for this group
+                results[f'{sensitive_val}'] = {
+                    'true_positive': 0,
+                    'false_positive': 0,
+                    'true_negative': 0,
+                    'false_negative': 0,
+                }
+                continue
+
+            group_predictions = predictions[mask]
+            group_actuals = actuals[mask]
+
+            # Compute confusion matrix
+            tn, fp, fn, tp = confusion_matrix(
+                group_actuals, group_predictions, labels=[0, 1]
+            ).ravel()
+
+            results[f'{sensitive_val}'] = {
+                'true_positive': int(tp),
+                'false_positive': int(fp),
+                'true_negative': int(tn),
+                'false_negative': int(fn),
+            }
+
+        return results
+
+    @classmethod
+    def _compute_equalized_odds_score(cls, prediction_counts):
+        """Compute the equalized odds score from prediction counts."""
+        # Extract counts for both groups
+        true_group = prediction_counts['True']
+        false_group = prediction_counts['False']
+
+        # Compute TPR for each group
+        tpr_true = true_group['true_positive'] / max(
+            1, true_group['true_positive'] + true_group['false_negative']
+        )
+        tpr_false = false_group['true_positive'] / max(
+            1, false_group['true_positive'] + false_group['false_negative']
+        )
+
+        # Compute FPR for each group
+        fpr_true = true_group['false_positive'] / max(
+            1, true_group['false_positive'] + true_group['true_negative']
+        )
+        fpr_false = false_group['false_positive'] / max(
+            1, false_group['false_positive'] + false_group['true_negative']
+        )
```

Comment:
Since we're doing the same operation multiple times, could we put it in a for loop?

Also, could you let a comment describing how tpr_true is calculated? I couldn't find a reference for the normalization.
```
tpr_true = true_group['true_positive'] / max(
            1, true_group['true_positive'] + true_group['false_negative']
        )
```

Reply:
True positive rate is also called [sensitivity](https://en.wikipedia.org/wiki/Sensitivity_and_specificity).